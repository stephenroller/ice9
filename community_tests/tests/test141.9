# mutual recursion										succeed

var i : int

forward a (p : int)

proc b (p : int)
	if i > 4 ->
		return;
	fi

	i := i + 1;	
	write "b";

	a(p);
end

proc a (p : int)
	write "a";
	
	b(p);
end

i := 0;
b(0);
i := 0;
a(0);