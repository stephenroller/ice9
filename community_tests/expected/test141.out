Loading file: generated/test141.tm
b
a
b
a
b
a
b
a
b
a
a
b
a
b
a
b
a
b
a
b
a
Number of instructions executed = 1084
